OEL-2710: Add editor to simple rich text format.

AaronGilMartinez opened the pull request:
## OPENEUROPA-[Insert ticket number here]

### Description

[Insert description here]

### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

AaronGilMartinez committed:
OEL-2710: Modify simple_rich_text editor.

AaronGilMartinez committed:
OEL-2761: Modify Publication test.

AaronGilMartinez committed:
OEL-2710: Move Social share funtion to trait, and News test to JS.

AaronGilMartinez committed:
OEL-2710: Move Person to JS.

AaronGilMartinez committed:
OEL-2710: Move event test.

AaronGilMartinez committed:
OEL-2710: Move project.

AaronGilMartinez committed:
OEL-2710: Clean code, remove unwanted change.

AaronGilMartinez committed:
OEL-2710: Add update.

AaronGilMartinez committed:
OEL-2710: Remove media button and restore tags.

drishu commented on config/install/filter.format.simple_rich_text.yml:
```diff
@@ -11,6 +13,18 @@ filters:
+    status: true
+    weight: 10
+    settings: {  }
+  oe_oembed_filter:
```
the supported tags dont include images so this should not work but it because of the order of the plugins; also the requirements didnt ask for this so please remove

AaronGilMartinez replied:
Removed

drishu replied:
you forgot to remove the oe_oembed dependency

AaronGilMartinez replied:
Done

drishu commented on config/install/filter.format.simple_rich_text.yml:
```diff
@@ -11,6 +13,18 @@ filters:
     weight: -10
     settings:
-      allowed_html: '<a href hreflang> <em> <strong>'
+      allowed_html: '<a href hreflang> <em> <strong> <ul> <li> <ol> <blockquote>'
```
the requirement asked not to change the supported tags

AaronGilMartinez replied:
Removed

drishu commented on config/post_updates/00033_editor_simple_rich_text/editor.editor.simple_rich_text.yml:
```diff
@@ -0,0 +1,41 @@
+            - DrupalLink
+            - DrupalUnlink
+        -
+          name: Lists
```
keep to existing supported tags

AaronGilMartinez replied:
Aligned

drishu commented on config/post_updates/00033_editor_simple_rich_text/filter.format.simple_rich_text.yml:
```diff
@@ -0,0 +1,30 @@
+    status: true
+    weight: -10
+    settings:
+      allowed_html: '<a href hreflang> <em> <strong> <ul> <li> <ol> <blockquote>'
```
after fixing the installed config remember to fix also these update ones

AaronGilMartinez replied:
changes applied on both, install and update

drishu commented on tests/src/ExistingSiteJavascript/EventTest.php:
```diff
@@ -43,10 +43,10 @@ public function testEventType(): void {
-      'admin/structure/taxonomy/manage/event_type/overview',
-      'admin/structure/taxonomy/manage/event_type/add',
-    ], 'editor');
+    $this->drupalGet('admin/structure/taxonomy/manage/event_type/overview');
```
the previous code did a lot, why did we change this ?

AaronGilMartinez replied:
The trait checks status code and couldn't find a way to do that in JS tests. An option to keep the code would be try to find common assertions for both types Webassert/JSWebassert and modifying the trait.

drishu replied:
its used in ContactFormTest and seems to be green

AaronGilMartinez replied:
As we saw ContactFormTest is using Functional tests and there is no quick way to get the status.

drishu commented on tests/src/ExistingSiteJavascript/EventTest.php:
```diff
@@ -133,7 +135,6 @@ public function testCreateEvent(): void {
     // Assert user without permission can't create events.
     $this->drupalGet('node/add/oe_sc_event');
     $assert_session->pageTextContains('You are not authorized to access this page.');
-    $assert_session->statusCodeEquals(403);
```
why remove ?

AaronGilMartinez replied:
Status code can't be asserted with JSWebAssert. Couldn't find a way to check it for JS tests.

drishu replied:
its used in ContactFormTest and it seems to be green

AaronGilMartinez replied:
As we saw ContactFormTest is using Functional tests and there is no quick way to get the status.

drishu commented on tests/src/ExistingSiteJavascript/NewsTest.php:
```diff
@@ -79,16 +78,18 @@ public function testCreateNews() {
     // Assert that editors have access to the Simple/Rich text formats.
     $assert_session->pageTextNotContains('This field has been disabled because you do not have sufficient permissions to edit it.');
+    $introduction = $page->findField('Introduction');
+    $this->assertEquals('simple_rich_text', $this->getWysiwigTextFormat($introduction));
```
Lets improve these text editor tests by adding some allowed markup and some not allowed markup

AaronGilMartinez replied:
Added some HTML to the content and checked what is rendered and what is not in this test. Not sure if we could add a trait to check this across different test, not only for the simple_rich_text also for the rich text.

drishu replied:
Your intuition is right, I did mean everywhere this text editor is used not just here. But I think it would be more efficient to have a separate test for the editor, and then we already test the presence.
Since time on the ticket is scarce, lets create a ticket for testing all editors.

AaronGilMartinez replied:
Created this https://citnet.tech.ec.europa.eu/CITnet/jira/browse/OEL-2821

drishu commented on tests/src/ExistingSiteJavascript/ProjectTest.php:
```diff
@@ -87,40 +104,37 @@ public function testCreateProject() {
     $page->pressButton('Create organisation');
-
-    // Summary, objective, impacts and achievements and milestones.
-    $field = $page->findField('Summary');
```
we remove it but we didnt put anything in its place

AaronGilMartinez replied:
Just reordered the assertions, with Javascript tests requires to scroll to the section, the removed code its above.

AaronGilMartinez committed:
OEL-2794: Supported tags and filter.

AaronGilMartinez committed:
OEL-2710: HTML allowed tags in test.

AaronGilMartinez committed:
OEL-2710: Remove dependency.

brummbar commented on tests/src/ExistingSiteJavascript/ProjectTest.php:
```diff
@@ -213,8 +227,10 @@ public function testCreateProject() {
     $page->fillField('Reference', '');
-    $page->pressButton('Remove');
-    $page->pressButton('Remove');
+    $page->find('css', 'input[value="Remove"]')->press();
```
Not sure about this. Why this change? Inputs are still considered buttons.

AaronGilMartinez replied:
The pressButton method looks for button tags
![image](https://github.com/openeuropa/oe_showcase/assets/7264392/b32f3db1-6e91-47bf-8b02-7aa5c2ae3e4e)

With JS enabled those are Ajax and are replaced with inputs
![image](https://github.com/openeuropa/oe_showcase/assets/7264392/04e38621-b928-43a9-8884-c5e2c84617f2)

Not sure if there is another way to target them

AaronGilMartinez committed:
OEL-2710: Fix grump error.

drishu committed:
OEL-2710: Fix tests.

AaronGilMartinez committed:
OEL-2710: Project edition fix.

drishu committed:
OEL-2710: Fix phpcs warning.

brummbar committed:
OEL-2710: Move testing of editor format in a dedicated test.

We want to avoid using JS tests as much as possible, and since the
editor is tied to a format, it's enough to test just one instance of
the editor in one content type.

brummbar committed:
OEL-2710: Test the two formats.
